cognite/client/_api/diagrams.py

```diff
@@ -74,9 +79,21 @@ def _process_file_ids(ids: Union[Sequence[int], int, None], external_ids: Union[
         else:
             raise TypeError("external_ids must be str or list of str")
 
+        if isinstance(file_references, FileReference):
+            file_references = [file_references]
+        elif (
+            isinstance(file_references, list)
+            and all(isinstance(fr, FileReference) for fr in file_references)
+            or file_references is None
```

Comment:
Could do the `None` check first as a tiny optimization?

Also, what do you think about making the all-isinstance-check into an utility function? 🤔

Reply:
I dont think it would be an optimization, but some refactoring might be an idea

Reply:
Oh, true that, didn’t spot the isinstance check that’s done first 😊